### PR TITLE
added responsible disclosure policy

### PR DESCRIPTION
**Pull request by kevinhammond**
First try at responsible disclosure based on ETH approach

### CODE REVIEW

File: policy/security/README.md [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+## Security Issues
+
+Possible security concerns should be notified to: XX
+
+All security concerns will be evaluated.  If a vulnerability is deemed
+to exist, then a fix will be developed and released.  All care will be
+taken not to expose the nature of the vulnerability, both during
+code development and as part of the release.  This may include using a private
+fork for the  development of the fix, including the fix as part of another code commit,
+taking appropriate care when writing code comments/descriptions, silent fixes, or other techniques.
```
[the remark this answers is not in the document]
Comment (coot) [reply policy/security/README.md:19]: Well it says

> This _may_ include using a private fork

which arguably could be made into _MAY_ :grin:

**Comment by abailly-iohk** [comment]
This policy from the Apache Software foundation seems to provide everything we would want: https://www.apache.org/security/committers.html

File: policy/security/README.md [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+
+
+## Responsible Disclosure
+
+
+If we silently fix a vulnerability and a hard fork is needed to enable the fix then:
+
+- we will publish the details about the vulnerability 4-8 weeks after the hard fork
```
[the remark this answers is not in the document]
Comment (kevinhammond) [reply policy/security/README.md:27]: This came more or less directly from the Ethereum policy

https://geth.ethereum.org/docs/vulnerabilities/vulnerabilities

A hard fork will reveal the vulnerability, but the reality is that the purpose of patches can often be obscured.  We have fixed a number of issues in the node this way (without revealing the purpose of the fix for some time after the event).

The complication with a hard fork is that all actors must participate.  The nice thing is that the patch is applied uniformly after the hard fork event.

File: policy/security/README.md [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+Maintaining the security of the Cardano blockchain is a paramount
+concern.  Consistent with other blockchain projects, we will adopt a
+policy of responsible notification, responsible development, and responsible disclosure of
+security patches.  Given the open nature of Cardano development, this will require particular
+care over how security issues are handled by code maintainers.
+
+
+## Security Issues
+
+Possible security concerns should be notified to: XX
```
[the remark this answers is not in the document]
Comment (kevinhammond) [reply policy/security/README.md:12]: > Would you have some references to share on this? It seems to me there's some contradiction between hiding security issues and releasing changes containing fixes, but I am far from being a security expert.

This came more or less directly from the Ethereum policy

https://geth.ethereum.org/docs/vulnerabilities/vulnerabilities

Added a reference

File: policy/security/README.md [hunk 4]
```diff
@@ -1,6 +1,37 @@
 # Security
 
+Maintaining the security of the Cardano blockchain is a paramount
+concern.  Consistent with other blockchain projects, we will adopt a
```
[the remark this answers is not in the document]
Comment (kevinhammond) [reply policy/security/README.md:4]: OK

File: policy/security/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinhammond) [reply policy/security/README.md:19]: It has to be taken on a case-by-case basis

File: policy/security/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevinhammond) [reply policy/security/README.md:27]: > I think it's important to say what it means to "silently fix a vulnerability".

Added a section

File: policy/security/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevinhammond) [reply policy/security/README.md:12]: > suggestion: put a default address and let's raise an IT ticket to create the address and route it to a bunch of people.

We will use the existing security@iohk.io

**Comment by kevinhammond** [comment]
> This policy from the Apache Software foundation seems to provide everything we would want: https://www.apache.org/security/committers.html

It's a good basis, but would need to be adapted.  I suggest we pass that on to the security team for them to action and to review against our current security policy.  This section was just supposed to be a short policy on responsible disclosure.

**Comment by kevinhammond** [comment]
If we could merge this version that would be good - that gives someone a basis for expanding the policy later...

**Review by abailly-iohk — APPROVED** [review]

**Pull request merged**